capstone/src/main/resources/static/event.js

```diff
@@ -28,108 +29,188 @@ function getAllEventsForSearch(data) {
     reviewTimeElement.className = 'timeago';
     reviewTimeElement.setAttribute('datetime', review.timestamp);
 
     const reviewTextElement = createElement(reviewContainer, 'p', review.text);
     reviewTextElement.className = 'review-text';
+
+    const reviewLikeElement = createElement(reviewContainer, 'button',  review.likes + ' Likes');
+    reviewLikeElement.addEventListener('click', () => {
+      likeReview(review.datastoreId).then((reviewLikes) => {
+        reviewLikeElement.innerText = reviewLikes + ' Likes';
+      });
+    });
+
+    if (review.individualEmail == userEmail) {
+      reviewAuthorFeatures(reviewContainer, reviewTextElement, review.datastoreId);
+    }
   })
 }
 
+/**
+ * Add delete and edit functionality for review's author
+ * @param reviewContainer Review's container
+ * @param reviewTextElement Container for review's text
+ * @param reviewId Review's datastore id
+ */
+function reviewAuthorFeatures(reviewContainer, reviewTextElement, reviewId) {
+  const deleteButton = createElement(reviewContainer, 'button', 'Delete');
+  deleteButton.addEventListener('click', () => {
+    deleteReview(reviewId);
+    reviewContainer.remove();
+  });
+
+  const editButton = createElement(reviewContainer, 'button', 'Edit');
+  editButton.addEventListener('click', () => {
+    if (editButton.innerText == 'Edit') {
+      reviewTextElement.contentEditable = true;
+      reviewTextElement.focus();
+      editButton.innerText = 'Done';
+    } else {
+      setReviewText(reviewId, reviewTextElement.innerText);
+      reviewTextElement.contentEditable = false;
+      editButton.innerText = 'Edit';
+    }
+  });
+}
+
 /**
  * Create new review to add to event's list
  * @param eventId Event's datastoreId
  * @param text Text content of Review
+ * @return Update list of reviews
  */
 async function newReview(eventId, text) {
   const params = new URLSearchParams();
   params.append('text', text);
   params.append('eventId', eventId);
-  //params.append('name', individual[0].firstName + ' ' + individual[0].lastName);
-  //Quick fix until create a new way to attach user to review
-  params.append('name', 'quick-fix');
   const response = await fetch('new-review', {method:'POST', body: params});
   const reviews = response.json();
   getEvents();
   return reviews;
+}
 
+/**
+ * Remove a review from event's list
+ * @param reviewId Review to remove's datastoreId
+ */
+function deleteReview(reviewId) {
+  const params = new URLSearchParams();
+  params.append('reviewId', reviewId);
+  fetch('delete-review', {method:'POST', body: params});
+  getEvents();
+}
+
+/**
+ * Add to review's like count
+ * @param reviewId Review's datastoreId
+ * @return Updated like count
+ */
+async function likeReview(reviewId) {
+  const params = new URLSearchParams();
+  params.append('reviewId', reviewId);
+  const response = await fetch('review-likes', {method:'POST', body: params});
+  const value = response.json();
+  getEvents();
```

Comment:
Instead of calling getEvents for all these functions, I think you can move this function to be called when the body is onload. This way the user would see a list of events when they enter the index.html and you also don't need to call it for every function.

Reply:
oh yeah that's a good idea! #128